### ISSUE TITLE
API Attribute Option Check Is Case Sensitive, But Add Attribute Option Check Is Case Insensitive

### ISSUE THREAD

**Issue by mhelm75** [akeneo/pim-community-dev#12100]
:bug: **I'm reporting a Bug** :bug:

Akeneo v4.0.24

I'm working on a fairly large Akeneo implementation that contains a job the operates nightly to automatically generate new products in Akeneo from another system using API calls. The job also generates new attribute option values in case they don't already exist in Akeneo.

Recently I ran into a situation where my content team created an option with code "2_Year_Limited_Manufacturer_Warranty" in anticipation for new products to be processed with this value. The products came through with the value "2_year_limited_manufacturer_warranty" and caused the product update to fail. However, when the team went in to add the missing value Akeneo gave the error message "This value is already used".

I think it makes sense to not want 2 versions of a code where one has capital letters and the other doesn't, but if that's the case then the API should automatically assign "2_Year_Limited_Manufacturer_Warranty" when given "2_year_limited_manufacturer_warranty"

**Comment by sugalvojau** [comment]
We have the very similar issue on our side.

An API request `{{url}}/api/rest/v1/products?search={"camera_brand":[{"operator":"IN","value":["alpina"]}]}` is failing because the code of the actual attribute option is `ALPINA`, not `alpina`. 

The failure of the previous request would not be that big issue if a user would be able to simply add the new value `alpina` while `ALPINA` already exists. But Akeneo UI gives the error in such case, adding the new value is not possible.

In short, pushing changes to Akeneo is case-insensitive. While pulling the changes back from Akeneo (e.g. via search in API or in other ways internally) is case sensitive.

I have been able to reproduce the same issue on http://demo.akeneo.com/ so it seems like it is not the issue of an old version or alike.

**Comment by PauliusMacernis** [comment]
Is it soved in version 3.0.82 (ticket PIM-9282 , https://github.com/akeneo/pim-community-dev/releases/tag/v3.0.82 )? I have no ability to check it so cannot confirm it works for now.
